src/Button/Button.component.tsx

```diff
@@ -352,42 +372,22 @@ const StyledButton = styled.button<StyledButtonProps>`
             : buttonStyles.base}
 
     /* Sizing */
-    padding: ${({ $size, circular, iconOnly }: StyledButtonProps) =>
-        iconOnly
-            ? '0'
-            : `0 ${
-                  circular
-                      ? dimensions[$size].circularPadding
-                      : dimensions[$size].padding
-              }rem`};
-    font-size: ${({ $size }) => dimensions[$size].fontSize}rem;
-    height: ${({ $height }) => $height}rem;
-    ${({ $size, minWidth, iconOnly, $height, block }: StyledButtonProps) =>
-        iconOnly
-            ? css`
-                  // Make it a square
-                  width: ${$height}rem;
-              `
-            : block
+    ${({ $fontSize, $height, padding }) => css`
+        padding: ${padding};
+        font-size: ${rem($fontSize)};
```

Comment:
I'd like to convert these to `@xstyled/system` utility functions, but I hit two issues:
 - My prop is named `$fontSize` instead of `fontSize`
 - My fontSize is in rem units. xstyled/styled-components automatically assumes numerics are `px`

There are solutions, but I punted on solving it.